// users.proto
syntax = "proto3";

package users;

option go_package = "github.com/yourusername/library/users";

service UsersService {
  rpc Register (RegisterRequest) returns (UserResponse) {}
  rpc GetUser (GetUserRequest) returns (UserResponse) {}
  rpc UpdateUser (UpdateUserRequest) returns (UserResponse) {}
  rpc Authenticate (AuthRequest) returns (AuthResponse) {}
}

message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
  string full_name = 4;
  string address = 5;
}

message GetUserRequest {
  string user_id = 1;
}

message UpdateUserRequest {
  string user_id = 1;
  string email = 2;
  string full_name = 3;
  string address = 4;
}

message AuthRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse {
  bool success = 1;
  string user_id = 2;
  string token = 3;
  string message = 4;
}

message UserResponse {
  string user_id = 1;
  string username = 2;
  string email = 3;
  string full_name = 4;
  string address = 5;
  string created_at = 6;
}